File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/postgresql/StreamingInsertSingle.kt
package dev.typr.foundations.docs.postgresql

import dev.typr.kotlinfoundations.*

@Suppress("unused")
class StreamingInsertSingle {

    //start
    // Insert a list of strings using COPY
    fun insertNames(names: List<String>, tx: Transactor): Long {
        return streamingInsert
            .of("COPY users(name) FROM STDIN", 1000, names.iterator(), PgTypes.text.pgText())
            .transact(tx)
    }
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/postgresql/StreamingInsertSingle.scala
package dev.typr.foundationssc.docs.postgresql

import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object StreamingInsertSingle:

  //start
  // Insert a list of strings using COPY
  def insertNames(names: Iterator[String], tx: Transactor): Long =
    streamingInsert
      .of("COPY users(name) FROM STDIN", 1000, names, PgTypes.text.pgText())
      .transact(tx)
  //stop
